File: src/main/java/tech/vhrd/vhl/psi/VhlFile.java
package tech.vhrd.vhl.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import tech.vhrd.vhl.VhlFileType;
import tech.vhrd.vhl.VhlLanguage;
import org.jetbrains.annotations.NotNull;

public class VhlFile extends PsiFileBase {
    public VhlFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, VhlLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return VhlFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "Vhl File";
    }
}

File: src/main/kotlin/tech/vhrd/vhl/psi/VhlFile.kt
package tech.vhrd.vhl.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import tech.vhrd.vhl.VhLanguage
import tech.vhrd.vhl.VhlFileType

class VhlFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, VhLanguage) {
    override fun getFileType(): FileType = VhlFileType
    override fun toString(): String = "Vhl file"
}